code/modules/surgery/organs/internal/heart/heart_anomalock.dm

```diff
@@ -0,0 +1,173 @@
+	Overcharges itself to grants the user a second wind upon receiving fatal damage."
+	icon_state = "anomalock_heart"
+	///Cooldown for the activation of the organ
+	var/survival_cooldown = 5 MINUTES
+	///Stores current time of when the organ was last activated
+	var/last_activation = -5 MINUTES //We should be off cooldown even if world.time is 0
+	///Maximum amount of time the organ will remain "active"
+	var/active_duration = 30 SECONDS
+	///If our organ is currently active
+	var/active = FALSE
+	///The lightning effect on our mob when the implant is active
+	var/mutable_appearance/lightning_overlay
+
+	//---- Anomaly core variables:
+	///The core item the organ runs off.
+	var/obj/item/assembly/signaler/anomaly/core
+	///Accepted types of anomaly cores.
+	var/list/accepted_anomalies = list(/obj/item/assembly/signaler/anomaly/flux)
+	///If this one starts with a core in.
+	var/prebuilt = FALSE
+	///If the core is removable once socketed.
+	var/core_removable = TRUE
+
+/obj/item/organ/internal/heart/cybernetic/anomalock/on_mob_insert(mob/living/carbon/organ_owner, special, movement_flags)
+	. = ..()
+	if(!core)
+		return
+	lightning_overlay = mutable_appearance(icon = 'icons/effects/effects.dmi', icon_state = "lightning")
+	organ_owner.add_overlay(lightning_overlay)
+	addtimer(CALLBACK(organ_owner, TYPE_PROC_REF(/atom, cut_overlay), lightning_overlay), 3 SECONDS)
+	playsound(organ_owner, 'sound/items/eshield_recharge.ogg', 40)
+	ADD_TRAIT(organ_owner, TRAIT_SHOCKIMMUNE, REF(src))
+	organ_owner.AddElement(/datum/element/empprotection, EMP_PROTECT_SELF|EMP_PROTECT_CONTENTS)
+	organ_owner.apply_status_effect(/datum/status_effect/stabilized/yellow, src)
+	RegisterSignal(organ_owner, SIGNAL_ADDTRAIT(TRAIT_CRITICAL_CONDITION), PROC_REF(activate_survival))
+	RegisterSignal(organ_owner, COMSIG_ATOM_EMP_ACT, PROC_REF(on_emp_act))
+
+/obj/item/organ/internal/heart/cybernetic/anomalock/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	if(!core)
+		return
+	UnregisterSignal(organ_owner, SIGNAL_ADDTRAIT(TRAIT_CRITICAL_CONDITION))
+	REMOVE_TRAIT(organ_owner, TRAIT_SHOCKIMMUNE, REF(src))
+	organ_owner.RemoveElement(/datum/element/empprotection, EMP_PROTECT_SELF|EMP_PROTECT_CONTENTS)
+	organ_owner.remove_status_effect(/datum/status_effect/stabilized/yellow)
+	tesla_zap(source = organ_owner, zap_range = 20, power = 2.5e5, cutoff = 1e3)
+	qdel(src)
+
+/obj/item/organ/internal/heart/cybernetic/anomalock/attack(mob/living/target_mob, mob/living/user, params)
+	if(target_mob == user && istype(target_mob) && core)
+		if(DOING_INTERACTION(user, "implanting"))
+			return
+		user.balloon_alert(user, "This will hurt...")
+		to_chat(user, span_userdanger("You see tendrils opening your ribcage"))
+		if(!do_after(user, 5 SECONDS, interaction_key = "implanting"))
+			return ..()
+		playsound(target_mob, 'sound/weapons/slice.ogg', 50, TRUE)
+		user.temporarilyRemoveItemFromInventory(src, TRUE)
+		Insert(user)
+		user.apply_damage(50, BRUTE, BODY_ZONE_CHEST)
```

Comment:
```suggestion
		user.apply_damage(100, BRUTE, BODY_ZONE_CHEST)
```

it'd be kinda funny if self-implanting it caused the crit buff to auto-trigger. makes for more of a downside to self-implanting than 5 units of salicylic acid